src/parse/gast_builder.ts

```diff
@@ -640,3 +643,82 @@ export function findClosingOffset(
         throw new Error("INVALID INPUT TEXT, UNTERMINATED PARENTHESIS")
     }
 }
+
+export function deserializeGrammar(
+    grammar: ISerializedGast[],
+    terminals: ITerminalNameToConstructor
+): IProduction[] {
+    return map(grammar, production =>
+        deserializeProduction(production, terminals)
+    )
+}
+
+export function deserializeProduction(
+    node: ISerializedGastAny,
+    terminals: ITerminalNameToConstructor
+): IProduction {
+    switch (node.type) {
```

Comment:
There is a very naive GAST visitor in Chevrotain.
Did you prefer a switch case intentionally ?

Reply:
Are you talking about the visitor defined in "src/parse/grammar/gast/gast_visitor_public.ts"? I saw it but it seems like that visitor goes from `IProduction -> any` whereas I would need a visitory that goes from `ISerializedGast -> IProduction`. Let me know if this is incorrect.

Reply:
Yeah you are correct 😄